aspnetcore/signalr/java-client.md

```diff
@@ -36,29 +37,45 @@ If using Maven, add the following lines inside the `<dependencies>` element of y
 
 To establish a `HubConnection`, the `HubConnectionBuilder` should be used. The hub URL and log level can be configured while building a connection. Configure any required options by calling any of the `HubConnectionBuilder` methods before `build`. Start the connection with `start`.
 
-[!code-java[Build hub connection](java-client/sample/src/main/java/Chat.java?range=17-20)]
+[!code-java[Build hub connection](java-client/sample/src/main/java/Chat.java?range=16-17)]
 
 ## Call hub methods from client
 
 A call to `send` invokes a hub method. Pass the hub method name and any arguments defined in the hub method to `send`.
 
-[!code-java[send method](java-client/sample/src/main/java/Chat.java?range=31)]
+[!code-java[send method](java-client/sample/src/main/java/Chat.java?range=28)]
 
 ## Call client methods from hub
 
 Use `hubConnection.on` to define methods on the client that the hub can call. Define the methods after building but before starting the connection.
 
-[!code-java[Define client methods](java-client/sample/src/main/java/Chat.java?range=22-24)]
+[!code-java[Define client methods](java-client/sample/src/main/java/Chat.java?range=19-21)]
+
+## Add logging
+
+The SignalR Java client uses the [SLF4J](https://www.slf4j.org/) library for logging. It's a high-level logging API that allows users of the library to chose their own specific logging implementation by bringing in a specific logging dependency. The following code snippet shows how to use `java.util.logging` with the SignalR Java client.
+
+```gradle
+implementation 'org.slf4j:slf4j-jdk14:1.7.25'
+```
+
+If you don't configure logging in your dependencies, SLF4J loads a default no-operation logger with the following warning message:
+
+```
+SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
+SLF4J: Defaulting to no-operation (NOP) logger implementation
+SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
+```
+
+This can safely be ignored.
 
 ## Known limitations
 
-This is an early preview release of the Java client. There are many features that aren't supported yet. The following gaps are being worked on for future releases:
+This is a preview release of the Java client. There are some features that aren't supported yet. The following gaps are being worked on for future releases:
```

Comment:
Outside of docs that are specifically identified as "roadmap" docs, we aren't supposed to say anything that could be construed as a promise for future development. Consider rewording this slightly, something like:

> This is a preview release of the Java client. Some features aren't supported:
>
> * Only the JSON protocol is supported.
> * Only the WebSockets transport is supported.
> * Streaming isn't supported.

Ordinarily you might add something like this:

> For information about plans for future development, see the [ASP.NET Core Roadmap](https://github.com/aspnet/AspNetCore/wiki/Roadmap).

But in this case it doesn't look like the roadmap has any of these details, so that link won't be very helpful here.